src/param.rs

```diff
@@ -227,92 +304,303 @@ impl AudioParamProcessor {
     }
 
     fn tick(&mut self, ts: f64, dt: f64, count: usize) -> &[f32] {
-        // store incoming automation events in sorted queue
+        // println!("> tick - ts: {}, dt: {}, count: {}", ts, dt, count);
+        // handle incoming automation events in sorted queue
+        // let events: Vec<AutomationEvent> = self.receiver.try_iter().collect();
+
+        // cf. https://www.w3.org/TR/webaudio/#computation-of-value
+        // 1. paramIntrinsicValue will be calculated at each time, which is either the
+        // value set directly to the value attribute, or, if there are any automation
+        // events with times before or at this time, the value as calculated from
+        // these events. If automation events are removed from a given time range,
+        // then the paramIntrinsicValue value will remain unchanged and stay at its
+        // previous value until either the value attribute is directly set, or
+        // automation events are added for the time range.
         for event in self.receiver.try_iter() {
+            // param intrisic value must be updated from the set_value call
+            // but we don't want to insert these events in the queue
+            if event.event_type == AutomationType::SetValue {
+                let current_value = self.shared_value.load() as f32;
```

Comment:
Are you sure this is correct? Should we not take the value from the `event`?

Reply:
Yes that's correct, because `shared_value` is immediately updated when `set_value` is called, and before the `SetValue` event is even dispatched. This allows to make sure the value stays consistant in the control thread before the next tick is called.

But you are right, as the value of the event is exactly the same here, it's more logical and surely better to use it instead of accessing the `AtomicF64`.